### PR TITLE
fix for using query fields in generated-geometry 199

### PR DESCRIPTION
**Pull request by imranrajjad**
query created by generated-geometry was using all available fields

https://github.com/geosolutions-it/C105-2018-EMSA-VDS/issues/199

### CODE REVIEW

File: src/community/generated-geometries/src/main/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategy.java [hunk 1]
```diff
@@ -230,23 +232,42 @@ public Filter convertFilter(FeatureTypeInfo info, Filter filter) {
...
-                            .filter(
-                                    propertyDescriptor ->
-                                            !propertyDescriptor
-                                                    .getName()
-                                                    .toString()
-                                                    .equals(configuration.geomAttributeName))
-                            .map(propertyDescriptor -> propertyDescriptor.getName().toString())
-                            .collect(Collectors.toList());
+            // no fields were sent, use all fields excluding geom field
+            if (query.getPropertyNames() == null) {
```
Comment (nmco) [inline src/community/generated-geometries/src/main/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategy.java]: Well done, later improve this by also checking is empty.

Comment (nmco) [reply src/community/generated-geometries/src/main/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategy.java]: Actually here if no field is request nothing shoudl probably be done., this shoudl be manually tested.

**Comment by nmco** [comment]
My comments shoudl be reviewed and if need applied on another PR.

File: src/community/generated-geometries/src/main/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategy.java (hunk 1 above)
Comment (imranrajjad) [reply src/community/generated-geometries/src/main/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategy.java]: No field scenario appears from Layer Config page when bounds are calculated using data and not SRS, in that case the property names come empty

**Pull request merged**